### ISSUE TITLE
e2e tests are working only on pristine clusters

### ISSUE THREAD

**Issue by jpkrohling** [jaegertracing/jaeger-operator#355]
The e2e tests are currently working only when executed against a clean cluster. 

Steps to reproduce:

1. `minikube start`
1. Run `make e2e-tests` -- it should pass
1. Run it again -- it fails already in the smoke tests. In the operator logs, this can be seen:

```
time="2019-04-03T12:41:29Z" level=error msg="failed to apply the changes" error="no matches for kind \"Elasticsearch\" in version \"logging.openshift.io/v1alpha1\"" execution="2019-04-03 12:41:29.381660248 +0000 UTC" instance=agent-as-sidecar namespace=smoke-smoke-sidecar-1554295279
```

Interestingly, I see the ES operator deployed, which shouldn't be there for Kubernetes:

```console
$ kubectl get deployments --all-namespaces
NAMESPACE                          NAME                       READY     UP-TO-DATE   AVAILABLE   AGE
kube-system                        coredns                    2/2       2            2           14m
kube-system                        default-http-backend       1/1       1            1           14m
kube-system                        nginx-ingress-controller   1/1       1            1           14m
openshift-logging                  elasticsearch-operator     1/1       1            1           4m24s
smoke-smoke-daemonset-1554295154   jaeger-operator            1/1       1            1           95s
```

**Comment by jpkrohling** [comment]
Disabling the goal `e2e-tests-self-provisioned-es` made it work again. 

cc @pavolloffay

**Comment by pavolloffay** [comment]
> Interestingly, I see the ES operator deployed, which shouldn't be there for Kubernetes:

The es-operator deploy goal runs always. It's deployed via make. I am not sure how hard it would be to  deploy it in go test or alternatively find out if we are on OCP in makefile

**Comment by pavolloffay** [comment]
@jpkrohling I was able to repetitively run e2e tests on minikube without problems.

**Comment by jpkrohling** [comment]
This was fixed by #366. Looks like my "Fixes" comment there didn't close this one :/
